tests/framework/behaviors/CacheableBehaviorTest.php

```diff
@@ -0,0 +1,221 @@
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        $this->initializeApplicationMock();
+        $this->initializeWidgetMocks();
+    }
+
+    public function testDefaultCache()
+    {
+        $this->assertEquals('cache', $this->simpleWidget->cache);
+    }
+
+    public function testDefaultDuration()
+    {
+        $this->assertEquals(60, $this->simpleWidget->cacheDuration);
+    }
+
+    public function testDefaultDependency()
+    {
+        $this->assertNull($this->simpleWidget->cacheDependency);
+    }
+
+    public function testDefaultKey()
+    {
+        $this->assertNull($this->simpleWidget->cacheKey);
+    }
+
+    public function testDefaultKeyVariations()
+    {
+        $this->assertEquals([], $this->simpleWidget->cacheKeyVariations);
+    }
+
+    public function testDefaultEnabled()
+    {
+        $this->assertTrue($this->simpleWidget->cacheEnabled);
+    }
+
+    public function testWidgetIsRunWhenCacheIsEmpty()
+    {
+        $this
+            ->simpleWidget
+            ->expects($this->once())
+            ->method('run')
+        ;
+
+        $contents = $this->simpleWidget->test();
+        $this->assertEquals('contents', $contents);
+    }
+
+    public function testWidgetIsNotRunWhenCacheIsNotEmpty()
+    {
+        $this->simpleWidget->cacheDuration = 0;
+        $this
+            ->simpleWidget
+            ->expects($this->once())
+            ->method('run')
+        ;
+
+        for ($counter = 0; $counter <= 42; $counter++) {
+            $this->assertEquals('contents', $this->simpleWidget->test());
+        }
+    }
+
+    public function testDynamicContent()
+    {
+        $this->dynamicWidget->cacheDuration = 0;
+        $this
+            ->dynamicWidget
+            ->expects($this->once())
+            ->method('run')
+        ;
+
+        for ($counter = 0; $counter <= 42; $counter++) {
+            $expectedContents = sprintf('<div>dynamic contents: %d</div>', $counter);
+            $this->assertEquals($expectedContents, $this->dynamicWidget->test());
+        }
+    }
+
+    /**
+     * Initializes a mock application.
+     *
+     */
+    private function initializeApplicationMock()
+    {
+        $this->mockApplication([
+            'components' => [
+                'cache' => [
+                    'class' => '\yii\caching\ArrayCache',
+                ],
+            ],
+            'params' => [
+                // Counter for dynamic contents testing.
+                'counter' => 0,
+            ],
+        ]);
+    }
+
+    /**
+     * Initializes mock widgets.
+     *
+     */
+    private function initializeWidgetMocks()
+    {
+        $this->simpleWidget = $this->getWidgetMock(SimpleCacheableWidget::className());
+        $this->dynamicWidget = $this->getWidgetMock(DynamicCacheableWidget::className());
+    }
+
+    /**
+     * Returns a widget mock.
+     *
+     */
+    private function getWidgetMock($widgetClass)
+    {
+        $widgetMock = $this
```

Comment:
Changed that. Still believe style is not the best possible.
1. It's inconsistent since methods are not aligned. Why do we make difference for the first method?
2. Placing semicolon on the same line with one of the methods is a bad idea: when we'd need to add another method at the end of the call chain, git diff will contain lines which were not effectively changed what will complicate the code-review process.